### PR TITLE
Catch on Card creation

### PR DESCRIPTION
**Pull request by kordianbruck**
When a card crashes during creation, this leaves the home screen in a loading state. By catching those exceptions, we at least don't block from other cards being displayed, as one card is having troubles.

Related: https://www.support.tum.de/otrs/index.pl?Action=AgentTicketZoom;TicketID=3131883

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/CardsRepository.kt [hunk 1]
```diff
@@ -74,8 +75,12 @@ class CardsRepository(private val context: Context) {
         }
 
         providers.forEach { provider ->
-            val cards = provider.getCards(cacheControl)
-            results.addAll(cards)
+            try {
+                val cards = provider.getCards(cacheControl)
+                results.addAll(cards)
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/CardsRepository.kt]: please add the reasoning of the PR as a comment

**Comment by thellmund** [comment]
@kordianbruck said that the crash is caused by CafeteriaManager, which is a little odd: we're already doing a null check in its getCards() method.  Rather than just putting a try-catch here, I'd rather investigate the underlying issue.

**Comment by kordianbruck** [comment]
@thellmund 
```
de.tum.in.tumcampus E/CardsRepository: java.lang.IllegalStateException: db.cafeteriaDao().getById(id) must not be null
    java.lang.IllegalStateException: db.cafeteriaDao().getById(id) must not be null
        at de.tum.in.tumcampusapp.component.ui.cafeteria.repository.CafeteriaLocalRepository.getCafeteria(CafeteriaLocalRepository.kt:45)
        at de.tum.in.tumcampusapp.component.ui.cafeteria.repository.CafeteriaLocalRepository.getCafeteriaNameFromId(CafeteriaLocalRepository.kt:30)
        at de.tum.in.tumcampusapp.component.ui.cafeteria.repository.CafeteriaLocalRepository.getCafeteriaWithMenus(CafeteriaLocalRepository.kt:24)
        at de.tum.in.tumcampusapp.component.ui.cafeteria.details.CafeteriaViewModel.getCafeteriaWithMenus(CafeteriaViewModel.kt:36)
        at de.tum.in.tumcampusapp.component.ui.cafeteria.controller.CafeteriaManager.getCafeteriaWithMenus(CafeteriaManager.java:84)
        at de.tum.in.tumcampusapp.component.ui.cafeteria.controller.CafeteriaManager.getCards(CafeteriaManager.java:60)
        at de.tum.in.tumcampusapp.component.ui.overview.CardsRepository.getCardsNow(CardsRepository.kt:79)
        at de.tum.in.tumcampusapp.component.ui.overview.CardsRepository.access$getCardsNow(CardsRepository.kt:24)
        at de.tum.in.tumcampusapp.component.ui.overview.CardsRepository$refreshCards$1.invoke(CardsRepository.kt:44)
        at de.tum.in.tumcampusapp.component.ui.overview.CardsRepository$refreshCards$1.invoke(CardsRepository.kt:24)
        at org.jetbrains.anko.AsyncKt$doAsync$1.invoke(Async.kt:140)
        at org.jetbrains.anko.AsyncKt$doAsync$1.invoke(Async.kt)
        at org.jetbrains.anko.AsyncKt$sam$Callable$892965c7.call(Async.kt)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:272)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
```

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/CardsRepository.kt (hunk 1 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/overview/CardsRepository.kt]: Done

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/onboarding/StartupActivity.java [hunk 2]
```diff
@@ -59,6 +63,11 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_startup);
 
+        // Only use Crashlytics if we are not compiling debug
+        if (!BuildConfig.DEBUG) {
+            Fabric.with(this, new Crashlytics());
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/onboarding/StartupActivity.java]: A try-catch around this might also solve #1134

**Comment by pfent** [comment]
underlying issue still needs to be fixed.
please open an issue for it

**Pull request merged**